### PR TITLE
added scalapb oneOf support

### PR DESCRIPTION
**Pull request by ilyadoroshenko**
https://github.com/wix-private/server-infra/issues/14909

### CODE REVIEW

File: chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala [hunk 1]
```diff
@@ -619,10 +619,21 @@ trait TransformerMacros extends TransformerConfigSupport with MappingMacros with
...
-      resolveTransformerBodyTreeFromAccessorsMapping(srcPrefixTree, accessorsMapping, From, To, config)
-        .map(_ ++ overridesMapping)
+      def mkTransformer(From: Type, srcPrefixTree: Tree) = {
+        val accessorsMapping = resolveAccessorsMapping(From, notOverridenTargets, config)
+
+        resolveTransformerBodyTreeFromAccessorsMapping(srcPrefixTree, accessorsMapping, From, To, config)
+          .map(_ ++ overridesMapping)
+      }
+
+      /** Proto oneOf to sealed trait enum support
```
Comment (d12frosted) [inline chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala:630]: 💯 for linking that issue 😸 you are saving our future minds 😸

File: chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala [hunk 2]
```diff
@@ -619,10 +619,21 @@ trait TransformerMacros extends TransformerConfigSupport with MappingMacros with
...
+       *  @see https://github.com/wix-private/server-infra/issues/14909
+       */
+      if (From.baseClasses.exists(_.fullName.contains("scalapb.GeneratedOneof")))
```
Comment (d12frosted) [inline chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala:633]: Cool, this works when converting from `oneOf` to domain types. Looks like converting vice versa would be as easy, right? 😸

**Review by d12frosted — APPROVED** [review]
AMAZING! 🎉

File: chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala (hunk 2 above)
Comment (ilyadoroshenko) [reply chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala:633]: yeah, should be straightforward
I'm wondering if Piotr was ok with merging this 💩 upstream 🤣

**Pull request merged**